src/audio/module_adapter/module/volume/volume_hifi3_with_peakvol.c

```diff
@@ -21,348 +21,9 @@ LOG_MODULE_DECLARE(volume_hifi3, CONFIG_SOF_LOG_LEVEL);
-	ae_f16x4 *out = (ae_f16x4 *)audio_stream_wrap(sink, (char *)audio_stream_get_wptr(sink)
-						      + bsink->size);
-	const int channels_count = audio_stream_get_channels(sink);
-	const int inc = sizeof(ae_f32x2);
-	int samples = channels_count * frames;
-	ae_f32x2 temp;
-	ae_f32x2 *peakvol = (ae_f32x2 *)cd->peak_vol;
-
-	/* Set peakvol(which stores the peak volume data 4 times) as circular buffer */
-	memset(peakvol, 0, sizeof(ae_f32) * channels_count * 4);
-	AE_SETCBEGIN1(cd->peak_vol);
-	AE_SETCEND1(cd->peak_vol  + channels_count * 4);
-
-	/** to ensure the adsress is 8-byte aligned and avoid risk of
-	 * error loading of volume gain while the cd->vol would be set
-	 * as circular buffer
-	 */
-	if (cd->copy_gain)
-		vol_store_gain(cd, channels_count);
-
-	buf = (ae_f32x2 *)cd->vol;
-	buf_end = (ae_f32x2 *)(cd->vol + channels_count * 4);
-	vol = buf;
-
-	/* Set buf as circular buffer */
-	AE_SETCBEGIN0(buf);
-	AE_SETCEND0(buf_end);
-
-	while (samples) {
-		m = audio_stream_samples_without_wrap_s16(source, in);
-		n = MIN(m, samples);
-		m = audio_stream_samples_without_wrap_s16(sink, out);
-		n = MIN(m, n);
-		inu = AE_LA64_PP(in);
-		for (i = 0; i < n; i += 4) {
-			/* load first two volume gain */
-			AE_L32X2_XC(volume0, vol, inc);
-
-			/* load second two volume gain */
-			AE_L32X2_XC(volume1, vol, inc);
-
-#if COMP_VOLUME_Q8_16
-			/* Q8.16 to Q9.23 */
-			volume0 = AE_SLAI32S(volume0, 7);
-			volume1 = AE_SLAI32S(volume1, 7);
-#elif COMP_VOLUME_Q1_23
-			/* No need to shift, Q1.23 is OK as such */
-#else
-#error "Need CONFIG_COMP_VOLUME_Qx_y"
-#endif
-			/* Load the input sample */
-			AE_LA16X4_IP(in_sample, inu, in);
-			/* calculate the peak volume*/
-			AE_L32X2_XC1(temp, peakvol, 0);
-			temp = AE_MAXABS32S(AE_SEXT32X2D16_32(in_sample), temp);
-			AE_S32X2_XC1(temp, peakvol, inc);
-			/* calculate the peak volume*/
-			AE_L32X2_XC1(temp, peakvol, 0);
-			temp = AE_MAXABS32S(AE_SEXT32X2D16_10(in_sample), temp);
-			AE_S32X2_XC1(temp, peakvol, inc);
-			/* Multiply the input sample */
-			out_sample0 = AE_MULFP32X16X2RS_H(volume0, in_sample);
-			out_sample1 = AE_MULFP32X16X2RS_L(volume1, in_sample);
-
-			/* Q9.23 to Q1.31 */
-			out_sample0 = AE_SLAI32S(out_sample0, 8);
-			out_sample1 = AE_SLAI32S(out_sample1, 8);
-
-			/* store the output */
-			out_sample = AE_ROUND16X4F32SSYM(out_sample0, out_sample1);
-			AE_SA16X4_IP(out_sample, outu, out);
-		}
-		AE_SA64POS_FP(outu, out);
-		samples -= n;
-		in = audio_stream_wrap(source, in);
-		out = audio_stream_wrap(sink, out);
-		bsource->consumed += VOL_S16_SAMPLES_TO_BYTES(n);
-		bsink->size += VOL_S16_SAMPLES_TO_BYTES(n);
-	}
-	for (i = 0; i < channels_count; i++) {
-		m = MAX(cd->peak_vol[i], cd->peak_vol[i + channels_count]);
-		m = MAX(m, cd->peak_vol[i + channels_count * 2]);
-		m = MAX(m, cd->peak_vol[i + channels_count * 3]);
-		cd->peak_regs.peak_meter[i] = m;
-	}
-}
-#endif /* CONFIG_FORMAT_S16LE */
-
-#else /* HIFI3 VERSION */
+#if CONFIG_COMP_PEAK_VOL
```

Comment:
at least we could select peak-volume specific files in CMakeLists.txt?

Reply:
Sorry i was on vacation that time, and this PR is merged. Then i will fix these in the next PR.

Reply:
@andrula-song no worries, doing that in a separate PR would be good!